cohorts/survival.py

```diff
@@ -72,59 +77,89 @@ def plot_kmf(df,
         if is_median:
             label_suffix += " (median)"
         default_label_with_condition = "%s > %s" % (condition_col, label_suffix)
-    else:
+        with_condition_label = with_condition_label or default_label_with_condition
+        no_condition_label = no_condition_label or default_label_no_condition
+        label_map = {False: no_condition_label,
+                     True: with_condition_label}
+        color_map = {False: no_condition_color,
+                     True: with_condition_color}
+    elif df[condition_col].dtype == 'O' or df[condition_col].dtype.name == "category":
```

Comment:
Actually, the `lambda` solution does work already, b/c tavi's `cols` logic is processed before this call to plot_kmf. Previously your lambda had to return a boolean but now can return a string or multiple categories. 

One thing I'd like to do is refactor this to not have the `no_condition_*`, `with_condition_*` etc. and instead take the color_map & label_map parameters directly. I wrote the function to support this but will implement it another day.